analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/TestHelper.cs

```diff
@@ -185,6 +185,13 @@ public static IMethodSymbol GetMethodSymbol(this (SyntaxTree, SemanticModel) tup
             return semanticModel.GetDeclaredSymbol(syntaxTree.GetMethod(name, skip));
         }
 
+        public static IFieldSymbol GetFieldSymbol(string fieldDefinition)
```

Comment:
I don't like this method, because it's doing something else than the other `GetXxx` methods above.

After removing the `ObjectConstraintTest`, this could go to the `ProgramStateTest`, as it will only be used there.
```suggestion
        public static IFieldSymbol CreateFieldSymbol(string field)
```